gb-cpu/src/registers/mod.rs

```diff
@@ -0,0 +1,139 @@
+pub mod area;
+
+use super::error::Error;
+use area::Area;
+use std::fmt;
+
+#[derive(Debug, Default)]
+pub struct Registers {
+    a: u8,
+    f: u8,
+    b: u8,
+    c: u8,
+    d: u8,
+    e: u8,
+    h: u8,
+    l: u8,
+    sp: u16,
+    pub pc: u16,
+}
+
+impl fmt::Display for Registers {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Registers {{ a: {}, f: {}, b: {}, c: {}, d: {}, e: {}, h: {}, l: {}, sp: {}, cp: {} }}",
+            self.a,
+            self.f,
+            self.b,
+            self.c,
+            self.d,
+            self.e,
+            self.h,
+            self.l,
+            self.sp,
+            self.pc)
+    }
+}
+
+impl Registers {
+    pub fn next_pc(&mut self) {
+        self.pc = self.pc.wrapping_add(1);
+    }
+
+    pub fn read_u8(&self, area: Area) -> Result<u8, Error> {
+        match area {
+            Area::A => Ok(self.a),
+            Area::F => Ok(self.f),
+            Area::B => Ok(self.b),
+            Area::C => Ok(self.c),
+            Area::D => Ok(self.d),
+            Area::E => Ok(self.e),
+            Area::H => Ok(self.h),
+            Area::L => Ok(self.l),
+            _ => Err(Error::InvalidRegister(area)),
```

Comment:
This would never happen when you read a 8bits register

Reply:
Same, wasn't sure so played it safe, really open to deeply understand why it's impossible